src/sdg/producten/management/commands/check_broken_links.py

```diff
@@ -0,0 +1,307 @@
+import time
+from collections import defaultdict
+from itertools import chain
+from urllib import parse
+
+from django.core.management import BaseCommand
+
+import markdown
+import requests
+from bs4 import BeautifulSoup
+from zgw_consumers.concurrent import parallel
+
+from sdg.core.models import ProductFieldConfiguration
+from sdg.producten.models import BrokenLinks, LocalizedProduct, Product
+
+FIELD_NAMES_CONFIG = {
+    "input_fields": [
+        "product_titel_decentraal",
+        "product_valt_onder_toelichting",
+        "product_aanwezig_toelichting",
+    ],
+    "markdown_fields": [
+        "specifieke_tekst",
+        "vereisten",
+        "bewijs",
+        "procedure_beschrijving",
+        "kosten_en_betaalmethoden",
+        "uiterste_termijn",
+        "bezwaar_en_beroep",
+        "wtd_bij_geen_reactie",
+    ],
+    "urls_fields": [
+        "verwijzing_links",
+    ],
+    "decentrale_label": ["decentrale_procedure_label"],
+    "decentrale_link": ["decentrale_procedure_link"],
+}
+VALID_URL_ADAPTERS = ("tel:", "sms:", "mailto:")
+INVALID_URL_ADAPTERS = "geo:"
+SUCCES_STATUS_CODE = 200
+ANY_ERROR_STATUS_CODE = 420
+REDIRECT_STATUS_CODES = {301, 302, 303, 307, 308}
+SUCCESSFUL_STATUS_CODES = {200} | REDIRECT_STATUS_CODES
+REDIRECT_CYCLES_LIMIT = 10
+
+
+class Command(BaseCommand):
+    help = "Check for broken links in product fields and update the BrokenLink model."
+    product_dict = defaultdict(lambda: defaultdict(list))
+    url_set = set()
+    url_response_status_codes = defaultdict(int)
+    founded_broken_link_ids = []
```

Comment:
Please don't use class-level attributes (that are initialized).

If you find this is needed because multiple functions are writing or reading the same class-variable, then the functions are incorrect or you need to pass the data around as argument to the next function.

For example, your function `get_products_to_check` doesn't "get" products. It fills a variable with products. Just, make this function return the products instead of filling a class-level variable.

Reply:
I tried to fix this in https://github.com/maykinmedia/sdg-invoervoorziening/pull/1018/commits/adc06025a452bdece4d943240dc4dc302a80fc53 ✅